cips/cip-040.md

```diff
@@ -0,0 +1,118 @@
+| cip | 40 |
+| - | - |
+| title | Privval Interface Extension for Arbitrary Message Signing |
+| description | Extends the CometBFT privval interface to support signing arbitrary messages for offchain protocols. |
+| author | CHAMI Rachid ([@rach-id](https://github.com/rach-id)), Evan Forbes ([evan-forbes](https://github.com/evan-forbes)) |
+| discussions-to | <https://forum.celestia.org/t/cip-40-privval-interface-extension-for-arbitrary-message-signing/2102> |
+| status | Draft |
+| type | Standards Track |
+| category | Interface |
+| created | 2025-07-25 |
+
+## Abstract
+
+This CIP proposes extending the CometBFT privval.Message interface to support signing arbitrary messages. This enhancement enables validators to use their consensus keys and existing Key Management Systems (KMS) to sign offchain protocol messages, facilitating the implementation of high-throughput and low-latency gossiping protocols while maintaining backwards compatibility.
```

Comment:
proposition to have code in `these` blocks. If you agree, maybe we should change all occurrences and also the chainID, uniqueID, etc
```suggestion
This CIP proposes extending the CometBFT `privval.Message` interface to support signing arbitrary messages. This enhancement enables validators to use their consensus keys and existing Key Management Systems (KMS) to sign offchain protocol messages, facilitating the implementation of high-throughput and low-latency gossiping protocols while maintaining backwards compatibility.
```